components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,1802 +1,1177 @@
+  real(r8) :: fh2o(pcols)                   ! h2o flux to balance source from methane chemistry
+  real(r8) :: tmp_q     (pcols,pver)        ! tmp variable
+  real(r8) :: tmp_cldliq(pcols,pver)        ! tmp variable
+  real(r8) :: tmp_cldice(pcols,pver)        ! tmp variable
+  real(r8) :: tmp_t     (pcols,pver)        ! tmp variable
+  real(r8) :: ftem      (pcols,pver)        ! tmp variable
+  logical  :: labort                        ! abort flag
+  logical  :: l_tracer_aero, l_vdiff, l_rayleigh, l_gw_drag, l_ac_energy_chk
+  !-----------------------------------------------------------------------------
+  !-----------------------------------------------------------------------------
+  lchnk = state%lchnk
+  ncol  = state%ncol
+  nstep = get_nstep()
+  
+  call phys_getopts( l_tracer_aero_out      = l_tracer_aero      &
+                    ,l_vdiff_out            = l_vdiff            &
+                    ,l_rayleigh_out         = l_rayleigh         &
+                    ,l_gw_drag_out          = l_gw_drag          &
+                    ,l_ac_energy_chk_out    = l_ac_energy_chk    )
 
-    ! get nstep and zero array for energy checker
-    zero = 0._r8
-    nstep = get_nstep()
-    call check_tracers_init(state, tracerint)
+  ! Adjust the surface fluxes to reduce instabilities in near sfc layer
+  if (phys_do_flux_avg()) call flux_avg_run(state, cam_in,  pbuf, nstep, ztodt)
 
-!!== KZ_WCON
+  ! Validate the physics state.
+  if (state_debug_checks) call physics_state_check(state, name="before tphysac")
 
-    call check_qflx(state, tend, "PHYAC01", nstep, ztodt, cam_in%cflx(:,1))
+  ! Associate pointers with physics buffer fields
+  itim_old = pbuf_old_tim_idx()
 
-#ifndef MMF_FLUX_BYPASS
-    if(.not.use_qqflx_fixer) then 
+  call pbuf_get_field(pbuf, pbuf_get_index('DTCORE'), dtcore, start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
 
-       ! Check if latent heat flux exceeds the total moisture content of the
-       ! lowest model layer, thereby creating negative moisture.
-#ifdef MAML
-       call qneg4('TPHYSAC '       ,lchnk               ,ncol  ,ztodt ,               &
-            num_inst_atm,state%q(1,pver,1),state%rpdel(1,pver) ,cam_in%shf(:,:) ,         &
-            cam_in%lhf(:,:) , cam_in%cflx )
-#else
-       call qneg4('TPHYSAC '       ,lchnk               ,ncol  ,ztodt ,               &
-            state%q(1,pver,1),state%rpdel(1,pver) ,cam_in%shf ,         &
-            cam_in%lhf , cam_in%cflx )
-#endif 
+  call pbuf_get_field(pbuf, tini_idx, tini)
+  call pbuf_get_field(pbuf, qini_idx, qini)
+  call pbuf_get_field(pbuf, cldliqini_idx, cldliqini)
+  call pbuf_get_field(pbuf, cldiceini_idx, cldiceini)
 
-    end if 
-#endif
+  call pbuf_get_field(pbuf, pbuf_get_index('CLD'),cld,start=(/1,1,itim_old/),kount=(/pcols,pver,1/))
+  call pbuf_get_field(pbuf, pbuf_get_index('AST'),ast,start=(/1,1,itim_old/),kount=(/pcols,pver,1/))
 
-    call check_qflx(state, tend, "PHYAC02", nstep, ztodt, cam_in%cflx(:,1))
```

Comment:
`check_qflx` also goes away?

Reply:
This routine only adds diagnostic information to the history files. I left the call to this in tphysbc, but these calls here that bracket qneg4 seem unnecessary. There are better ways to assess the impact of qneg4 if we are interested.